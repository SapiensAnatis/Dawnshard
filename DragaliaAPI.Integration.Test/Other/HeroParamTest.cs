using System.Net;
using System.Net.Http.Json;
using DragaliaAPI.Photon.Shared.Models;
using Snapshooter.Xunit;

namespace DragaliaAPI.Integration.Test.Other;

public class HeroParamTest : TestFixture
{
    public HeroParamTest(CustomWebApplicationFactory factory, ITestOutputHelper outputHelper)
        : base(factory, outputHelper) { }

    [Fact]
    public async Task HeroParam_ReturnsData()
    {
        this.ImportSave();

        HttpResponseMessage httpResponse = await this.Client.GetAsync($"heroparam/{ViewerId}/1");

        httpResponse.StatusCode.Should().Be(HttpStatusCode.OK);

        List<HeroParam>? heroParams = await httpResponse.Content.ReadFromJsonAsync<
            List<HeroParam>
        >();
        Snapshot.Match(heroParams);

        // AI should not have shared skills
        heroParams!
            .Skip(1)
            .Should()
            .AllSatisfy(x =>
            {
                x.editSkillcharacterId1.Should().Be(0);
                x.editSkillcharacterId2.Should().Be(0);
                x.editSkillLv1.Should().Be(0);
                x.editSkillLv2.Should().Be(0);
            });
    }
}
