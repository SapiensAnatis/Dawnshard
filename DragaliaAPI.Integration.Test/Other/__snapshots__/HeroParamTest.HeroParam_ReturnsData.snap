[
  {
    "level": 100,
    "characterId": 10850402,
    "hp": 923,
    "attack": 586,
    "defence": 0,
    "exAbilityLv": 5,
    "exAbility2Lv": 5,
    "ability1Lv": 3,
    "ability2Lv": 3,
    "ability3Lv": 3,
    "skill1Lv": 4,
    "skill2Lv": 3,
    "burstAttackLv": 2,
    "hpPlusCount": 100,
    "attackPlusCount": 100,
    "comboBuildupCount": 1,
    "dragonId": 20050401,
    "dragonLevel": 120,
    "dragonAbility1Lv": 6,
    "dragonAbility2Lv": 6,
    "dragonSkill1Lv": 2,
    "dragonSkill2Lv": 0,
    "dragonHpPlusCount": 50,
    "dragonAttackPlusCount": 50,
    "weaponBodyId": 30860401,
    "weaponBodyBuildupCount": 80,
    "weaponBodySkillNo": 1,
    "weaponBodySkillLv": 2,
    "weaponBodyAbility1Lv": 2,
    "weaponBodyAbility2Lv": 2,
    "weaponPassiveAbilityIds": [],
    "weaponSkinId": 0,
    "abilityCrestId": 40050130,
    "abilityCrestBuildupCount": 50,
    "abilityCrestAbility1Lv": 3,
    "abilityCrestAbility2Lv": 3,
    "abilityCrestHpPlusCount": 50,
    "abilityCrestAttackPlusCount": 50,
    "abilityCrest2Id": 40050090,
    "abilityCrest2BuildupCount": 50,
    "abilityCrest2Ability1Lv": 3,
    "abilityCrest2Ability2Lv": 3,
    "abilityCrest2HpPlusCount": 50,
    "abilityCrest2AttackPlusCount": 50,
    "abilityCrest3Id": 40050084,
    "abilityCrest3BuildupCount": 50,
    "abilityCrest3Ability1Lv": 3,
    "abilityCrest3Ability2Lv": 3,
    "abilityCrest3HpPlusCount": 50,
    "abilityCrest3AttackPlusCount": 50,
    "abilityCrest4Id": 40040062,
    "abilityCrest4BuildupCount": 40,
    "abilityCrest4Ability1Lv": 3,
    "abilityCrest4Ability2Lv": 3,
    "abilityCrest4HpPlusCount": 50,
    "abilityCrest4AttackPlusCount": 50,
    "abilityCrest5Id": 40040028,
    "abilityCrest5BuildupCount": 40,
    "abilityCrest5Ability1Lv": 3,
    "abilityCrest5Ability2Lv": 3,
    "abilityCrest5HpPlusCount": 50,
    "abilityCrest5AttackPlusCount": 50,
    "plusHp": 0,
    "plusAtk": 0,
    "plusDef": 0,
    "relativeHp": 0.0,
    "relativeAtk": 0.0,
    "relativeDef": 0.0,
    "dragonRelativeDmg": 0.5,
    "dragonTime": 0.0,
    "position": 1,
    "aiType": 0,
    "isEnemyTarget": true,
    "isFriend": false,
    "dragonReliabilityLevel": 30,
    "dragonRelativeHp": 0.0,
    "dragonRelativeAtk": 0.0,
    "dragonRelativeDef": 0.0,
    "battleGroup": 0,
    "eventPassiveGrowList": [],
    "eventBoostDmg": 0.0,
    "editSkillcharacterId1": 10540402,
    "editSkillcharacterId2": 10550101,
    "editSkillLv1": 2,
    "editSkillLv2": 4,
    "abilityCrest6Id": 40090001,
    "abilityCrest6BuildupCount": 30,
    "abilityCrest6Ability1Lv": 3,
    "abilityCrest6Ability2Lv": 3,
    "abilityCrest6HpPlusCount": 40,
    "abilityCrest6AttackPlusCount": 40,
    "abilityCrest7Id": 40090007,
    "abilityCrest7BuildupCount": 30,
    "abilityCrest7Ability1Lv": 3,
    "abilityCrest7Ability2Lv": 3,
    "abilityCrest7HpPlusCount": 40,
    "abilityCrest7AttackPlusCount": 40,
    "originalPosition": 0,
    "relativeHpFort": 1.4250001,
    "relativeAtkFort": 1.39,
    "relativeDefFort": 0.0,
    "relativeHpAlbum": 0.145,
    "relativeAtkAlbum": 0.145,
    "relativeDefAlbum": 0.0,
    "dragonRelativeHpFort": 0.1,
    "dragonRelativeAtkFort": 0.105,
    "dragonRelativeDefFort": 0.0,
    "dragonRelativeHpAlbum": 0.06,
    "dragonRelativeAtkAlbum": 0.051999997,
    "dragonRelativeDefAlbum": 0.0,
    "talismanId": 50250402,
    "talismanAbilityId1": 340000069,
    "talismanAbilityId2": 340000108,
    "talismanAbilityId3": 0,
    "talismanAdditionalHp": 0,
    "talismanAdditionalAttack": 0
  },
  {
    "level": 80,
    "characterId": 10350402,
    "hp": 763,
    "attack": 497,
    "defence": 0,
    "exAbilityLv": 5,
    "exAbility2Lv": 5,
    "ability1Lv": 2,
    "ability2Lv": 2,
    "ability3Lv": 2,
    "skill1Lv": 3,
    "skill2Lv": 2,
    "burstAttackLv": 2,
    "hpPlusCount": 100,
    "attackPlusCount": 100,
    "comboBuildupCount": 0,
    "dragonId": 20050414,
    "dragonLevel": 100,
    "dragonAbility1Lv": 5,
    "dragonAbility2Lv": 5,
    "dragonSkill1Lv": 2,
    "dragonSkill2Lv": 0,
    "dragonHpPlusCount": 50,
    "dragonAttackPlusCount": 50,
    "weaponBodyId": 30360401,
    "weaponBodyBuildupCount": 80,
    "weaponBodySkillNo": 1,
    "weaponBodySkillLv": 2,
    "weaponBodyAbility1Lv": 2,
    "weaponBodyAbility2Lv": 2,
    "weaponPassiveAbilityIds": [
      1030405
    ],
    "weaponSkinId": 0,
    "abilityCrestId": 40050130,
    "abilityCrestBuildupCount": 50,
    "abilityCrestAbility1Lv": 3,
    "abilityCrestAbility2Lv": 3,
    "abilityCrestHpPlusCount": 50,
    "abilityCrestAttackPlusCount": 50,
    "abilityCrest2Id": 40050111,
    "abilityCrest2BuildupCount": 50,
    "abilityCrest2Ability1Lv": 3,
    "abilityCrest2Ability2Lv": 3,
    "abilityCrest2HpPlusCount": 50,
    "abilityCrest2AttackPlusCount": 50,
    "abilityCrest3Id": 40050084,
    "abilityCrest3BuildupCount": 50,
    "abilityCrest3Ability1Lv": 3,
    "abilityCrest3Ability2Lv": 3,
    "abilityCrest3HpPlusCount": 50,
    "abilityCrest3AttackPlusCount": 50,
    "abilityCrest4Id": 40040003,
    "abilityCrest4BuildupCount": 40,
    "abilityCrest4Ability1Lv": 3,
    "abilityCrest4Ability2Lv": 3,
    "abilityCrest4HpPlusCount": 50,
    "abilityCrest4AttackPlusCount": 50,
    "abilityCrest5Id": 40040083,
    "abilityCrest5BuildupCount": 40,
    "abilityCrest5Ability1Lv": 3,
    "abilityCrest5Ability2Lv": 3,
    "abilityCrest5HpPlusCount": 50,
    "abilityCrest5AttackPlusCount": 50,
    "plusHp": 0,
    "plusAtk": 0,
    "plusDef": 0,
    "relativeHp": 0.0,
    "relativeAtk": 0.0,
    "relativeDef": 0.0,
    "dragonRelativeDmg": 0.5,
    "dragonTime": 0.0,
    "position": 2,
    "aiType": 0,
    "isEnemyTarget": true,
    "isFriend": false,
    "dragonReliabilityLevel": 30,
    "dragonRelativeHp": 0.0,
    "dragonRelativeAtk": 0.0,
    "dragonRelativeDef": 0.0,
    "battleGroup": 0,
    "eventPassiveGrowList": [],
    "eventBoostDmg": 0.0,
    "editSkillcharacterId1": 0,
    "editSkillcharacterId2": 0,
    "editSkillLv1": 0,
    "editSkillLv2": 0,
    "abilityCrest6Id": 40090007,
    "abilityCrest6BuildupCount": 30,
    "abilityCrest6Ability1Lv": 3,
    "abilityCrest6Ability2Lv": 3,
    "abilityCrest6HpPlusCount": 40,
    "abilityCrest6AttackPlusCount": 40,
    "abilityCrest7Id": 40090001,
    "abilityCrest7BuildupCount": 30,
    "abilityCrest7Ability1Lv": 3,
    "abilityCrest7Ability2Lv": 3,
    "abilityCrest7HpPlusCount": 40,
    "abilityCrest7AttackPlusCount": 40,
    "originalPosition": 0,
    "relativeHpFort": 1.44,
    "relativeAtkFort": 1.405,
    "relativeDefFort": 0.0,
    "relativeHpAlbum": 0.145,
    "relativeAtkAlbum": 0.145,
    "relativeDefAlbum": 0.0,
    "dragonRelativeHpFort": 0.1,
    "dragonRelativeAtkFort": 0.105,
    "dragonRelativeDefFort": 0.0,
    "dragonRelativeHpAlbum": 0.06,
    "dragonRelativeAtkAlbum": 0.051999997,
    "dragonRelativeDefAlbum": 0.0,
    "talismanId": 50950403,
    "talismanAbilityId1": 340000009,
    "talismanAbilityId2": 340000076,
    "talismanAbilityId3": 0,
    "talismanAdditionalHp": 0,
    "talismanAdditionalAttack": 0
  },
  {
    "level": 80,
    "characterId": 10750405,
    "hp": 751,
    "attack": 510,
    "defence": 0,
    "exAbilityLv": 5,
    "exAbility2Lv": 5,
    "ability1Lv": 2,
    "ability2Lv": 2,
    "ability3Lv": 2,
    "skill1Lv": 3,
    "skill2Lv": 2,
    "burstAttackLv": 2,
    "hpPlusCount": 100,
    "attackPlusCount": 100,
    "comboBuildupCount": 0,
    "dragonId": 20050412,
    "dragonLevel": 100,
    "dragonAbility1Lv": 5,
    "dragonAbility2Lv": 5,
    "dragonSkill1Lv": 2,
    "dragonSkill2Lv": 0,
    "dragonHpPlusCount": 50,
    "dragonAttackPlusCount": 50,
    "weaponBodyId": 30760401,
    "weaponBodyBuildupCount": 80,
    "weaponBodySkillNo": 1,
    "weaponBodySkillLv": 2,
    "weaponBodyAbility1Lv": 2,
    "weaponBodyAbility2Lv": 2,
    "weaponPassiveAbilityIds": [],
    "weaponSkinId": 0,
    "abilityCrestId": 40050130,
    "abilityCrestBuildupCount": 50,
    "abilityCrestAbility1Lv": 3,
    "abilityCrestAbility2Lv": 3,
    "abilityCrestHpPlusCount": 50,
    "abilityCrestAttackPlusCount": 50,
    "abilityCrest2Id": 40050084,
    "abilityCrest2BuildupCount": 50,
    "abilityCrest2Ability1Lv": 3,
    "abilityCrest2Ability2Lv": 3,
    "abilityCrest2HpPlusCount": 50,
    "abilityCrest2AttackPlusCount": 50,
    "abilityCrest3Id": 40050133,
    "abilityCrest3BuildupCount": 50,
    "abilityCrest3Ability1Lv": 3,
    "abilityCrest3Ability2Lv": 3,
    "abilityCrest3HpPlusCount": 50,
    "abilityCrest3AttackPlusCount": 50,
    "abilityCrest4Id": 40040063,
    "abilityCrest4BuildupCount": 40,
    "abilityCrest4Ability1Lv": 3,
    "abilityCrest4Ability2Lv": 3,
    "abilityCrest4HpPlusCount": 50,
    "abilityCrest4AttackPlusCount": 50,
    "abilityCrest5Id": 40040003,
    "abilityCrest5BuildupCount": 40,
    "abilityCrest5Ability1Lv": 3,
    "abilityCrest5Ability2Lv": 3,
    "abilityCrest5HpPlusCount": 50,
    "abilityCrest5AttackPlusCount": 50,
    "plusHp": 0,
    "plusAtk": 0,
    "plusDef": 0,
    "relativeHp": 0.0,
    "relativeAtk": 0.0,
    "relativeDef": 0.0,
    "dragonRelativeDmg": 0.5,
    "dragonTime": 0.0,
    "position": 3,
    "aiType": 0,
    "isEnemyTarget": true,
    "isFriend": false,
    "dragonReliabilityLevel": 30,
    "dragonRelativeHp": 0.0,
    "dragonRelativeAtk": 0.0,
    "dragonRelativeDef": 0.0,
    "battleGroup": 0,
    "eventPassiveGrowList": [],
    "eventBoostDmg": 0.0,
    "editSkillcharacterId1": 0,
    "editSkillcharacterId2": 0,
    "editSkillLv1": 0,
    "editSkillLv2": 0,
    "abilityCrest6Id": 40090007,
    "abilityCrest6BuildupCount": 30,
    "abilityCrest6Ability1Lv": 3,
    "abilityCrest6Ability2Lv": 3,
    "abilityCrest6HpPlusCount": 40,
    "abilityCrest6AttackPlusCount": 40,
    "abilityCrest7Id": 40090002,
    "abilityCrest7BuildupCount": 30,
    "abilityCrest7Ability1Lv": 3,
    "abilityCrest7Ability2Lv": 3,
    "abilityCrest7HpPlusCount": 40,
    "abilityCrest7AttackPlusCount": 40,
    "originalPosition": 0,
    "relativeHpFort": 1.373,
    "relativeAtkFort": 1.348,
    "relativeDefFort": 0.0,
    "relativeHpAlbum": 0.145,
    "relativeAtkAlbum": 0.145,
    "relativeDefAlbum": 0.0,
    "dragonRelativeHpFort": 0.1,
    "dragonRelativeAtkFort": 0.105,
    "dragonRelativeDefFort": 0.0,
    "dragonRelativeHpAlbum": 0.06,
    "dragonRelativeAtkAlbum": 0.051999997,
    "dragonRelativeDefAlbum": 0.0,
    "talismanId": 50150404,
    "talismanAbilityId1": 340000008,
    "talismanAbilityId2": 340000058,
    "talismanAbilityId3": 0,
    "talismanAdditionalHp": 0,
    "talismanAdditionalAttack": 0
  },
  {
    "level": 90,
    "characterId": 10250402,
    "hp": 825,
    "attack": 573,
    "defence": 0,
    "exAbilityLv": 5,
    "exAbility2Lv": 5,
    "ability1Lv": 3,
    "ability2Lv": 3,
    "ability3Lv": 3,
    "skill1Lv": 3,
    "skill2Lv": 3,
    "burstAttackLv": 2,
    "hpPlusCount": 100,
    "attackPlusCount": 100,
    "comboBuildupCount": 0,
    "dragonId": 20050418,
    "dragonLevel": 100,
    "dragonAbility1Lv": 5,
    "dragonAbility2Lv": 5,
    "dragonSkill1Lv": 2,
    "dragonSkill2Lv": 0,
    "dragonHpPlusCount": 50,
    "dragonAttackPlusCount": 50,
    "weaponBodyId": 30260401,
    "weaponBodyBuildupCount": 80,
    "weaponBodySkillNo": 1,
    "weaponBodySkillLv": 2,
    "weaponBodyAbility1Lv": 2,
    "weaponBodyAbility2Lv": 2,
    "weaponPassiveAbilityIds": [],
    "weaponSkinId": 0,
    "abilityCrestId": 40050122,
    "abilityCrestBuildupCount": 50,
    "abilityCrestAbility1Lv": 3,
    "abilityCrestAbility2Lv": 3,
    "abilityCrestHpPlusCount": 50,
    "abilityCrestAttackPlusCount": 50,
    "abilityCrest2Id": 40050075,
    "abilityCrest2BuildupCount": 50,
    "abilityCrest2Ability1Lv": 3,
    "abilityCrest2Ability2Lv": 3,
    "abilityCrest2HpPlusCount": 50,
    "abilityCrest2AttackPlusCount": 50,
    "abilityCrest3Id": 40050084,
    "abilityCrest3BuildupCount": 50,
    "abilityCrest3Ability1Lv": 3,
    "abilityCrest3Ability2Lv": 3,
    "abilityCrest3HpPlusCount": 50,
    "abilityCrest3AttackPlusCount": 50,
    "abilityCrest4Id": 40040063,
    "abilityCrest4BuildupCount": 40,
    "abilityCrest4Ability1Lv": 3,
    "abilityCrest4Ability2Lv": 3,
    "abilityCrest4HpPlusCount": 50,
    "abilityCrest4AttackPlusCount": 50,
    "abilityCrest5Id": 40040104,
    "abilityCrest5BuildupCount": 40,
    "abilityCrest5Ability1Lv": 3,
    "abilityCrest5Ability2Lv": 3,
    "abilityCrest5HpPlusCount": 50,
    "abilityCrest5AttackPlusCount": 50,
    "plusHp": 0,
    "plusAtk": 0,
    "plusDef": 0,
    "relativeHp": 0.0,
    "relativeAtk": 0.0,
    "relativeDef": 0.0,
    "dragonRelativeDmg": 0.5,
    "dragonTime": 0.0,
    "position": 4,
    "aiType": 0,
    "isEnemyTarget": true,
    "isFriend": false,
    "dragonReliabilityLevel": 30,
    "dragonRelativeHp": 0.0,
    "dragonRelativeAtk": 0.0,
    "dragonRelativeDef": 0.0,
    "battleGroup": 0,
    "eventPassiveGrowList": [],
    "eventBoostDmg": 0.0,
    "editSkillcharacterId1": 0,
    "editSkillcharacterId2": 0,
    "editSkillLv1": 0,
    "editSkillLv2": 0,
    "abilityCrest6Id": 40090001,
    "abilityCrest6BuildupCount": 30,
    "abilityCrest6Ability1Lv": 3,
    "abilityCrest6Ability2Lv": 3,
    "abilityCrest6HpPlusCount": 40,
    "abilityCrest6AttackPlusCount": 40,
    "abilityCrest7Id": 40090007,
    "abilityCrest7BuildupCount": 30,
    "abilityCrest7Ability1Lv": 3,
    "abilityCrest7Ability2Lv": 3,
    "abilityCrest7HpPlusCount": 40,
    "abilityCrest7AttackPlusCount": 40,
    "originalPosition": 0,
    "relativeHpFort": 1.4430001,
    "relativeAtkFort": 1.418,
    "relativeDefFort": 0.0,
    "relativeHpAlbum": 0.145,
    "relativeAtkAlbum": 0.145,
    "relativeDefAlbum": 0.0,
    "dragonRelativeHpFort": 0.1,
    "dragonRelativeAtkFort": 0.105,
    "dragonRelativeDefFort": 0.0,
    "dragonRelativeHpAlbum": 0.06,
    "dragonRelativeAtkAlbum": 0.051999997,
    "dragonRelativeDefAlbum": 0.0,
    "talismanId": 50150404,
    "talismanAbilityId1": 340000029,
    "talismanAbilityId2": 340000056,
    "talismanAbilityId3": 0,
    "talismanAdditionalHp": 0,
    "talismanAdditionalAttack": 0
  }
]
